.github/workflows/web-service.yml

```
@@ -0,0 +1,106 @@
+name: Web Service Release
+
+on:
+  workflow_dispatch:
+  push:
+
+permissions:
+  contents: read
+
+env:
+  DENY_WARNINGS: false
+
+jobs:
+  rust-build:
+    runs-on: hetzner-pm
+    container:
+      image: xaynetci/yellow:v3
+    timeout-minutes: 10
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Setup job
+        uses: ./.github/actions/setup-job-docker
+        with:
+          rust: true
+
+      - name: Build artifacts
+        run: just build-web-service
+
+      - name: Download assets
+        run: just download-assets
+
+      - name: Prepare temp directory
+        working-directory: ${{ runner.temp }}
+        run: |
+          rm -rf ./web-service
+          mkdir -p web-service/assets
+
+      - name: Create archive
+        working-directory: ${{ runner.temp }}
+        run: |
+          cp ${GITHUB_WORKSPACE}/${{ env.RUST_WORKSPACE }}/target/release/web-api ./web-service/web-api
+          cp ${GITHUB_WORKSPACE}/${{ env.RUST_WORKSPACE }}/web-api/.env ./web-service/.env
+          cp ${GITHUB_WORKSPACE}/${{ env.RUST_WORKSPACE }}/web-api/Dockerfile ./web-service/Dockerfile
+          cp ${GITHUB_WORKSPACE}/${{ env.FLUTTER_WORKSPACE }}/example/assets/smbert_v0001/smbert-quantized.onnx ./web-service/assets/model.onnx
+          cp ${GITHUB_WORKSPACE}/${{ env.FLUTTER_WORKSPACE }}/example/assets/smbert_v0001/vocab.txt ./web-service/assets/vocab.txt
+          tar -cvf web-service.tar ./web-service
+
+      - name: Upload archive
+        uses: actions/upload-artifact@v3
+        with:
+          name: web-service
+          retention-days: 1
+          if-no-files-found: error
+          path: ${{ runner.temp }}/web-service.tar
+
+  docker-build:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    needs: [rust-build]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Download artifacts
+        id: artifacts
+        uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741 # v3.0.0
+        with:
+          name: web-service
+
+      - name: Create docker image
+        run: |
+          tar -xvf web-service.tar
+          cd ./web-service
+          TAG="$(git rev-parse --short HEAD)"
+          if [[ "${GITHUB_REF#refs/heads/}" != "" ]]; then
+            #TAG="$(echo "branch-${GITHUB_REF#refs/heads/}" | tr '[:upper:]' '[:lower:]')"
+            TAG="branch-${GITHUB_REF#refs/heads/}"
+          fi
```

:question: I'm not sure why you originally used trim.

You should be able to get the branch name using `git rev-parse --abbrev-ref HEAD`, which will be `HEAD` if there is no branch name. 

So following should be a bit more clean:

```suggestion
          TAG="branch-$( git rev-parse --abbrev-ref HEAD)"
          if [[ "${TAG}" == "branch-HEAD" ]]; then
            TAG="$(git rev-parse --short HEAD)"
          fi
```

The difference between branch and commit hash is also normally very clear, so if we don't want to manually test on the tag starting with `branch-` we probably could just drop the `branch`.

---

oh, that was something that I just copied from prev solution, your version is better, thanks! :)